workspaces/ui-v2/src/pages/docs/EndpointRootPage.tsx

```diff
@@ -105,6 +110,80 @@ export const EndpointRootPage: FC<
   const unremoveEndpoint = () =>
     dispatch(documentationEditActions.unremoveEndpoint({ method, pathId }));
 
+  const isFieldRemoved = useCallback(
+    (fieldId: string) => {
+      if (userRemovedFields.includes(fieldId)) {
+        return 'root_removed';
+      } else if (allRemovedFields.has(fieldId)) {
+        return 'removed';
+      } else {
+        return 'not_removed';
+      }
+    },
+    [userRemovedFields, allRemovedFields]
+  );
```

Comment:
This totally works, but I wonder why pass down a function rather than just an `removedFields` array / set (or something similar)?  My first thought maybe this was a performance optimisation, but this will be called for every field regardless, so that's probably not right. Curious to know!

Reply:
I think there's the same number of calls per field? It's just the consumer of this specifies the behavior. Not a huge amount of reasoning why this over passing in the removedFields array / set - so feel free to update later on

Reply:
Fair enough, good to know whether you feel passionately or not. 

I don't see any risk or adverse performance effects from it (especially as we're also passing callbacks, so re-render optimisations are already limited somewhat), so no concern there. I've just tended to stay away from this pattern as it makes it a bit harder to debug by logging or inspecting props through the debugger. I do like not having to pass in weird data structures, though, so 🤷‍♂️. Lets live with it and find out!